src/wazuh_modules/syscollector/ports/portWindowsWrapper.h

```diff
@@ -0,0 +1,236 @@
+/*
+ * Wazuh SYSCOLLECTOR
+ * Copyright (C) 2015-2020, Wazuh Inc.
+ * November 3, 2020.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifndef _PORT_WINDOWS_WRAPPER_H
+#define _PORT_WINDOWS_WRAPPER_H
+#include <ws2ipdef.h>
+#include "iportWrapper.h"
+#include "sharedDefs.h"
+#include "stringHelper.h"
+
+const auto IPV6_BUFFER_ADDRESS_SIZE { 16 };
+
+static const std::map<int32_t, std::string> STATE_TYPE = 
+{
+    { MIB_TCP_STATE_ESTAB,                 "established"    },
+    { MIB_TCP_STATE_SYN_SENT,              "syn_sent"       },
+    { MIB_TCP_STATE_SYN_RCVD,              "syn_recv"       },
+    { MIB_TCP_STATE_FIN_WAIT1,             "fin_wait1"      },
+    { MIB_TCP_STATE_FIN_WAIT2,             "fin_wait2"      },
+    { MIB_TCP_STATE_TIME_WAIT,             "time_wait"      },
+    { MIB_TCP_STATE_CLOSED,                "close"          },
+    { MIB_TCP_STATE_CLOSE_WAIT,            "close_wait"     },
+    { MIB_TCP_STATE_LAST_ACK,              "last_ack"       },
+    { MIB_TCP_STATE_LISTEN,                "listening"      },
+    { MIB_TCP_STATE_CLOSING,               "closing"        },
+    { MIB_TCP_STATE_DELETE_TCB,            "delete_tcp"     }
+};
+
+static const std::map<pid_t, std::string> SYSTEM_PROCESSES = 
+{
+    { 0,                                   "System Idle Process"   },
+    { 4,                                   "System"                 },
+};
+
+struct PortTables
+{
+    std::unique_ptr<MIB_TCPTABLE_OWNER_PID []> tcp;
+    std::unique_ptr<MIB_TCP6TABLE_OWNER_PID []> tcp6;
+    std::unique_ptr<MIB_UDPTABLE_OWNER_PID []> udp;
+    std::unique_ptr<MIB_UDP6TABLE_OWNER_PID []> udp6;
+};
+
+class WindowsPortWrapper final : public IPortWrapper
+{
+    std::string m_protocol;
```

Comment:
make const as much as possible as they are init on the initializer list on ctor and returned by const methods.